CI: intermittent codecov coverage report failure

<!--

This issue tracker is a best-effort forum for users and customers to report bugs.

Be sure to not include any sensitive information. Sensitive information should __NOT__ be included in this issue.

-->

**Bug description**:
The codecov job in the CI intermittently fails with the following error:
```
Run bash <(curl -s https://codecov.io/bash) -F unittests
  bash <(curl -s https://codecov.io/bash) -F unittests
  shell: /usr/bin/bash -e {0}
  env:
    CI_WAIT_FOR_OK_SECONDS: 60
    CI_MAX_ITERATIONS_THRESHOLD: 60
    CI_CLIENT_CONCURRENT_CONNECTIONS: 1
    CI_MAX_WAIT_FOR_POD_TIME_SECONDS: 60
    CI_MIN_SUCCESS_THRESHOLD: 1
    OSM_HUMAN_DEBUG_LOG: true
    GOROOT: /opt/hostedtoolcache/go/1.16.9/x64
/dev/fd/63: line 1: $'{\r': command not found
/dev/fd/63: line 2: type:: command not found
/dev/fd/63: line 3: message:: command not found
/dev/fd/63: line 4: syntax error near unexpected token `}'
/dev/fd/63: line 4: ` }'
Error: Process completed with exit code 2.
```

**Affected area (please mark with X where applicable)**:
- CI System                  [X]

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -146,7 +146,9 @@ jobs:
         run: make go-test-coverage
       - name: Upload Coverage
         if: ${{ success() }}
-        run: bash <(curl -s https://codecov.io/bash) -F unittests
+        uses: codecov/codecov-action@v2
+        with:
+          flags: unittests
 
   scenarios_tests:
     name: Test various Envoy + SMI configuration scenarios

